modules/persistent-storage-csi-vsphere-install-issues.adoc

```diff
@@ -0,0 +1,27 @@
+// Module included in the following assemblies:
+//
+// persistent-storage-csi-vsphere.adoc
+//
+
+[id="persistent-storage-csi-vsphere-install-issues_{context}"]
+= Removing an upstream vSphere CSI Operator Driver
+
+If you have installed an upstream vSphere Container Storage Interface (CSI) Operator Driver, and then try to update your cluster to {product-title} 4.10 or later, {product-title} prevents the update.
+
+To correct this issue and update the cluster, a cluster admin must:
+
+. Delete the upstream vSphere CSI driver (VMware vSphere Container Storage Plug-in) Deployment and Daemonset objects.
+. Delete the configmap and secret objects that were installed previously with the upstream vSphere CSI Operator Driver.
+. Delete the upstream vSphereCSI driver `CSIDriver` object:
++
+[output, terminal]
+----
+~ $ oc delete CSIDriver csi.vsphere.vmware.com
+----
++
+[output, terminal]
+----
+csidriver.storage.k8s.io "csi.vsphere.vmware.com" deleted
+----
+
+The {product-title} update restarts automatically after you have completed removing the upstream vSphere CSI Operator Driver.
```

Comment:
@gnufied is it possible; that a customer 'partially' remove components; and the OSO installs our driver; and you get into a mixed state (but still end up degraded)? 

Would the product logs be able to help us narrow in on what objects/resources this is happening on / with?

Reply:
@gnufied @sferich888 In addition,4.x upgrade to 4.10 if there was a storageclass named "thin-csi" existed before upgrade, storage cluter operator degraded .https://bugzilla.redhat.com/show_bug.cgi?id=2055014 From my side, we add a check(Important reminding) here is better: 
```
Before upgrade confirm there is no storageclass name `thin-csi` 
```

Reply:
That sounds like a bug (I will confirm it tomorrow morning), so lets not add anything about `thin-csi` storageclass in documentation yet.

@sferich888 it is possible but possibility is somewhat remote, because we are checking for `CSIDriver` AND `CSINode` object and hence if something was left over, we will detect it and block installation of our own driver.

Reply:
@lpettyjo the documentation does not reflect wording I requested and is actually incorrect.